### PR TITLE
refactor(batch): refine sort agg

### PR DESCRIPTION
**Pull request by StrikeW**
## What's changed and what's your intention?
Remove the next chunk offset returned by aggregator for sort agg

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
previous PR #2084

### CODE REVIEW

**Comment by StrikeW** [comment]
@xiangjinwu PTAL

File: src/expr/src/vector_op/agg/general_sorted_grouper.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ impl EqGroups {
     }
 
     pub fn advance_offset(&mut self) {
-        if self.limit == 0 {
+        if (self.limit == 0) || (self.offset + self.limit >= self.indices.len()) {
```
Comment (xiangjinwu) [inline src/expr/src/vector_op/agg/general_sorted_grouper.rs:72]: Any existing code that uses `EqGroups` could potentially set a limit larger than `self.len()`? Or is it just for protection against future misuse?

**Review by xiangjinwu — COMMENTED** [review]
rest lgtm

File: src/expr/src/vector_op/agg/general_sorted_grouper.rs (hunk 1 above)
Comment (StrikeW) [reply src/expr/src/vector_op/agg/general_sorted_grouper.rs:72]: It is for protection.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**